### ISSUE TITLE
Inheritance is broken

### ISSUE THREAD

**Issue by mnn** [anttiviljami/mock-json-schema#1]
I am trying it with the `express-ts-mock` example (changed path to API definition, running it via `ts-node`, changed port, everything else should be same).

The API should be correct, I verified it in <code>swagger-ui-watcher</code> and all properties marked as required are visible as required (marked with red star, required flag is inherited along with the properties).

```yaml

openapi: 3.0.0
info:
  title: Sample API
  description: xxx
  version: 0.1.9

paths:
  /users:
    get:
      summary: xxx
      operationId: getUsers
      responses:
        '200':
          description: xxx
          content:
            application/json:
              schema:
                type: array
                items:
                  $ref: '#/components/schemas/B'

components:
  schemas:
    A:
      type: object
      title: A
      required:
        - g
      properties:
        g:
          type: string
          example: "G"
        h:
          type: number
          example: 1248
    B:
      allOf:
        - $ref: '#/components/schemas/A'
        - type: object
          title: B
          required:
            - i
          properties:
            i:
              type: string
              example: 'I'
```

<br>
How I tried it:

```js
$ curl http://localhost:9090/users
[{"g":"G","h":1248},{"i":"I"}]
```

The response is invalid. Both objects are invalid, because fields `i` and `g` are required, yet in the first object `i` is missing and in the second object `g` is missing.

Node: v8.9.4
package: 1.5.4

**Comment by anttiviljami** [comment]
Kinda embarrasing. I should've read the spec before implementing this. Sorry! 🙈

**Comment by mnn** [comment]
Oh, that seems like `allOf` is acting same way as `anyOf`? I might be reading it wrong, I am having a long day at work :weary:. I am very new to openapi (working with it for like two weeks without prior experience). I have to ask a team leader if I can try to fix it. But even if he agrees, I could get to it earliest by mid/end of the week.

No need to apologize, this is an open source project, people can't expect perfect experience for free :slightly_smiling_face:. And I have to say, response times here are amazing :smile:.

**Comment by anttiviljami** [comment]
Apparently I took `allOf` to mean it expects every schema to be present in the array. I'll just fix this then. :) If I haven't got around to it before long, please feel free to submit a PR!

**Comment by anttiviljami** [comment]
Should be fixed with v1.0.5 ! Let me know if you have any issues!
